base/linalg/transpose.jl

```diff
@@ -0,0 +1,334 @@
+immutable Transpose{T,S<:AbstractMatrix} <: AbstractMatrix{T}
+    data::S
+    conjugated::Bool
+end
```

Comment:
When I took a stab last year, I used a type parameter for this: https://github.com/JuliaLang/julia/compare/mb/transpose#diff-bcf746315d997712c8ec7f12b6126378R14.  But I now think it'd be simpler to follow Tim's advice and use a combination of two types: a MappedArray and a Transpose.  This effectively encodes the information within the type system… but note that you need a plain MappedArray no matter what in order to keep `(A').'` as a view.  So you may as well keep the two concerns orthogonal.